keyboards/zinc/keymaps/via/keymap.c

```diff
@@ -0,0 +1,195 @@
+   * `-----------------------------------------'             `-----------------------------------------'
+   */
+    [_ADJUST] =  LAYOUT_ortho_4x12(
+        _______, RESET,   RGBRST,  _______, _______, _______,                   _______, QWERTY,  _______, _______,  _______, KC_INS,
+        _______, RGB_TOG, RGB_HUI, RGB_SAI, RGB_VAI, AG_NORM,                   AG_SWAP, KC_MINS, KC_EQL,  KC_PSCR, KC_SLCK, KC_PAUS,
+        RGB_RMOD,RGB_MOD, RGB_HUD, RGB_SAD, RGB_VAD, _______,                   _______, _______, _______, _______, KC_PGUP, _______,
+        _______, _______, _______, EISU,    EISU,    EISU,                      KANA,    KANA,    KANA,    KC_HOME, KC_PGDN, KC_END
+        ),
+
+    [_ADJUST2] =  LAYOUT_ortho_4x12(
+        _______, RESET,   RGBRST,  _______, _______, _______,                   _______, QWERTY,  _______, _______,  _______, KC_INS,
+        _______, RGB_TOG, RGB_HUI, RGB_SAI, RGB_VAI, AG_NORM,                   AG_SWAP, KC_MINS, KC_EQL,  KC_PSCR, KC_SLCK, KC_PAUS,
+        RGB_RMOD,RGB_MOD, RGB_HUD, RGB_SAD, RGB_VAD, _______,                   _______, _______, _______, _______, KC_PGUP, _______,
+        _______, _______, _______, EISU,    EISU,    EISU,                      KANA,    KANA,    KANA,    KC_HOME, KC_PGDN, KC_END
+        )
+};
+
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case EISU:
+            if (record->event.pressed) {
+                if(keymap_config.swap_lalt_lgui==false){
+                    register_code(KC_LANG2);
+                } else {
+                    SEND_STRING(SS_LALT("`"));
+                }
+            } else {
+                unregister_code(KC_LANG2);
+            }
+            return false;
+        case KANA:
+            if (record->event.pressed) {
+                if(keymap_config.swap_lalt_lgui==false){
+                    register_code(KC_LANG1);
+                } else {
+                    SEND_STRING(SS_LALT("`"));
+                }
+            } else {
+                unregister_code(KC_LANG1);
+            }
+            return false;
+        case RGBRST:
+#ifdef RGBLIGHT_ENABLE
+            if (record->event.pressed) {
+                eeconfig_update_rgblight_default();
+                rgblight_enable();
+                RGB_current_config = rgblight_config;
+            }
+#endif
+            break;
+    }
+    return true;
+}
+
+
+void matrix_init_user(void) {
+#ifdef RGBLIGHT_ENABLE
+    rgblight_init();
+    RGB_current_config = rgblight_config;
+#endif
+}
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+#ifdef RGBLIGHT_LAYERS
+    // Both layers will light up if both kb layers are active
+    rgblight_set_layer_state(1, layer_state_cmp(state, 1));
+    rgblight_set_layer_state(2, layer_state_cmp(state, 2));
+#endif
+    return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST2);
+}
```

Comment:
You want to update the layer state before you check the state. 

```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
    state = update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST2);
#ifdef RGBLIGHT_LAYERS
    // Both layers will light up if both kb layers are active
    rgblight_set_layer_state(1, layer_state_cmp(state, 1));
    rgblight_set_layer_state(2, layer_state_cmp(state, 2));
#endif
    return state;
}
```